src/snmalloc/mem/freelist_queue.h

```diff
@@ -145,11 +135,15 @@ namespace snmalloc
       Cb cb)
     {
       invariant();
-      SNMALLOC_ASSERT(front.load() != nullptr);
+
+      freelist::HeadPtr curr = domesticate_head(front.load());
+      if (curr == nullptr)
+        // First entry is still in progress of being added.
+        // Nothing to do.
+        return;
```

Comment:
`{}` please. :)

Also more seriously, argh, what a frustrating branch!

Reply:
Yeah, but I needed to not have encoded pointers needed in the representation for empty.  This was the simplist solution I could come up with.

I am a bit unsure that we should encode nullptr, and instead have a representation that doesn't encode null.  I would like to do this, but as a subsequent PR.